pydantic_ai_slim/pydantic_ai/profiles/__init__.py

```diff
@@ -65,6 +65,13 @@ class ModelProfile:
     This is currently only used by `OpenAIChatModel`, `HuggingFaceModel`, and `GroqModel`.
     """
 
+    supported_builtin_tools: frozenset[str] = field(default_factory=lambda: frozenset[str]())
```

Comment:
- Same as up, I don't want plain strs, either a Literal, or the builtin tool types themselves
- I think the default should be _all builtin tools_, not none. As I said in the comment that sketched out this implementation :)